src/bin/main.rs

```diff
@@ -179,10 +185,31 @@ fn main() {
         }
     };
 
-    let code_files = paths.code_files.as_ref().map(|files| {
-        PathUtil::list_all_files(&files[..])
-            .unwrap()
-            .into_iter()
+    if let Some(code_files) = &paths.code_files {
+        if let Some(code_paths) = &paths.code_paths {
+            if code_files.len() != code_paths.len() {
+                eprintln!(
+                    "If argument code_paths is given in the toml file, it must have as many elements as argument code_files",
+                );
+                return;
+            }
+        }
+    }
+    let code_files = paths.code_files.as_ref().map(|patterns| {
```

Comment:
I would suggest sticking to `code_files` for the closure argument as the newly introduced `patterns` made me stumble when reading this.

Reply:
It might also make sense to merge this preprocessing with the above length check code to reduce the number of times these `Option` values are unpacked, something like
```rust
let code_files = match (&path.code_files, &path.code_paths) {
  (Some(code_files), Some(code_paths)) => ...,
  (Some(code_files), None) => ...,
  (None, _) => Vec::new(),
};
```